### PR TITLE
Update RoomSummaryCard navigation links

### PR DESCRIPTION
**Pull request by germain-gg**
For https://github.com/vector-im/element-web/issues/25883

This PR partially implements the new room details designs.

Not included in this PR:
- tab navigation. Items only accessible from the right panel tab navigation have been added as normal menu items (files, people)
- widget section
- [tab navigation header](https://www.figma.com/file/wqgXVb7RVaIybfRsXdWEiL/Element-Web-Room-Header-%26-Details?node-id=1481%3A146866&mode=dev) in subviews

Screenshot 2023-11-15 at 15 51 02
Screenshot 2023-11-15 at 15 50 29

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Update RoomSummaryCard navigation links ([\#11812](https://github.com/matrix-org/matrix-react-sdk/pull/11812)). Contributed by @germain-gg.

### CODE REVIEW

**Review by weeman1337 — CHANGES_REQUESTED** [review]
The new sidebar looks really good :art: 

Should the widgets item not also be a new menu item? It does now look a bit out of place.

image

Also without using the new room header you can now no longer reach the member list. Designs use tabs here. Maybe as an intermediate solution we could add a „People“ item?

File: src/components/views/right_panel/RoomSummaryCard.tsx [hunk 1]
```diff
@@ -369,43 +371,52 @@ const RoomSummaryCard: React.FC<IProps> = ({ room, permalinkCreator, onClose, on
...
+
+            <Separator />
+
+            <ToggleMenuItem
+                Icon={FavouriteIcon}
+                label={_t("room|context_menu|favourite")}
+                checked={isFavorite}
+                onClick={() => tagRoom(room, DefaultTagID.Favourite)}
+            />
+            <MenuItem
```
[the remark this answers is not in the document]
Comment (kerryarchibald) [reply src/components/views/right_panel/RoomSummaryCard.tsx:431]: Screenshot 2023-11-14 at 16 02 26

Fixed in compound 0.8.1

**Comment by americanrefugee** [comment]
I created a [new disabled menu item state](https://www.figma.com/file/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?type=design&node-id=2999-18282&mode=design&t=wkBXeDJWuNL511WL-4
).

**Comment by kerryarchibald** [comment]
hey @weeman1337, this is ready for another pass. thanks!

**Review by weeman1337 — APPROVED** [review]
The disabled buttons are now looking disabled :+1: 

Still the widgets actions are having a „link“ style:

image

Is there another issue to migrate the widget actions to the new menu design?

**Review by t3chguy — APPROVED** [review]
Approving the yarn.lock change

**Comment by kerryarchibald** [comment]
@weeman1337 the widgets change is captured here, under the 'next iteration' epic https://github.com/vector-im/wat-internal/issues/77

**Pull request merged**